src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingNullCaseInSwitchCheck.java

```diff
@@ -0,0 +1,210 @@
+                log(ast, MSG_KEY);
+            }
+        }
+    }
+
+    /**
+     * Gets all case labels in the given switch AST node.
+     *
+     * @param detailAST the AST node representing {@code LITERAL_SWITCH}
+     * @return a list of all case labels in the switch
+     */
+    private static List<DetailAST> getAllCaseLabels(DetailAST detailAST) {
+        final List<DetailAST> caseLabels = new ArrayList<>();
+        DetailAST ast = detailAST.getFirstChild();
+        while (ast != null) {
+            // case group token may have several LITERAL_CASE tokens
+            TokenUtil.forEachChild(ast, TokenTypes.LITERAL_CASE, caseLabels::add);
+            ast = ast.getNextSibling();
+        }
+        return caseLabels;
+    }
+
+    /**
+     * Checks if the given case AST node has a null label.
+     *
+     * @param caseLabel the AST node representing {@code LITERAL_CASE}
+     * @return true if the case has {@code null} label, false otherwise
+     */
+    private static boolean hasLiteralNull(DetailAST caseLabel) {
+        return Optional.ofNullable(caseLabel.findFirstToken(TokenTypes.EXPR))
+                .map(exp -> exp.findFirstToken(TokenTypes.LITERAL_NULL))
+                .isPresent();
+    }
+
+    /**
+     * Checks if the given case AST node has a pattern variable declaration label.
+     * or record pattern definition label.
+     * In this case, if the switch selector evaluates to null,
+     * a {@code NullPointerException} will be thrown.
+     *
+     * @param caseLabel the AST node representing {@code LITERAL_CASE}
+     * @return true if case has a pattern in its label
+     */
+    private static boolean hasPatternCaseLabel(DetailAST caseLabel) {
+        return caseLabel.findFirstToken(TokenTypes.RECORD_PATTERN_DEF) != null
+               || caseLabel.findFirstToken(TokenTypes.PATTERN_VARIABLE_DEF) != null
+               || caseLabel.findFirstToken(TokenTypes.PATTERN_DEF) != null;
+    }
+
+    /**
+     * Checks if the given case contains a string in its label.
+     * In this case, if the switch selector evaluates to null,
+     * a {@code NullPointerException} will be thrown.
+     *
+     * @param caseLabel the AST node representing {@code LITERAL_CASE}
+     * @return true if switch block contains a string case label
+     */
+    private static boolean hasStringCaseLabel(DetailAST caseLabel) {
+        return hasStringLiteral(caseLabel) || hasStringConcatenatedExpression(caseLabel);
+    }
+
+    /**
+     * Checks if the case label has a string literal.
+     *
+     * @param caseLabel the AST node representing {@code LITERAL_CASE}
+     * @return true if the case label has a string literal
+     */
+    private static boolean hasStringLiteral(DetailAST caseLabel) {
+        return Optional.ofNullable(caseLabel.findFirstToken(TokenTypes.EXPR))
+                .map(exp -> exp.findFirstToken(TokenTypes.STRING_LITERAL))
+                .isPresent();
```

Comment:
We can suppress these checker violations. I can see a bit of value in them sometimes, but in cases like this, we would double the lines of code needed to do the same thing and decrease readability.

Reply:
done